clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -747,10 +757,12 @@ public void onResponse(final ClientResponse response) {
             long currentTimeMs = response.receivedTimeMs();
             OffsetCommitResponse commitResponse = (OffsetCommitResponse) response.responseBody();
             Set<String> unauthorizedTopics = new HashSet<>();
+            Set<Uuid> unknownTopicIds = new HashSet<>();
             boolean failedRequestRegistered = false;
             for (OffsetCommitResponseData.OffsetCommitResponseTopic topic : commitResponse.data().topics()) {
+                String topicName = metadata.topicNames().getOrDefault(topic.topicId(), topic.name());
                 for (OffsetCommitResponseData.OffsetCommitResponsePartition partition : topic.partitions()) {
-                    TopicPartition tp = new TopicPartition(topic.name(), partition.partitionIndex());
+                    TopicPartition tp = new TopicPartition(topicName, partition.partitionIndex());
```

Comment:
`topic.name()` will be empty if the version is earlier then `v10`, and `metadata` might not contain topic name for `topic.topicId()` if the topic has been recreated. this scenario is similar to #19964, and perhaps we should throw an exception if `offsets` can't retrieve `OffsetAndMetadata`.